pybamm/expression_tree/symbol.py

```diff
@@ -738,7 +757,13 @@ def _jac(self, variable):
         """
         raise NotImplementedError
 
-    def _base_evaluate(self, t=None, y=None, y_dot=None, inputs=None):
+    def _base_evaluate(
+        self,
+        t: float | None = None,
```

Comment:
derived classes (e.g. scalar) have different types than this base function, should be consistent